realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SyncedRealmTests.kt

```diff
@@ -232,6 +239,222 @@ class SyncedRealmTests {
+
+            assertEquals(1, realm.where<EmbeddedSimpleListParent>().count())
+            assertEquals(2, realm.where<EmbeddedSimpleChild>().count())
+        }
+
+        val user2: User = createNewUser()
+        val config2: SyncConfiguration = createDefaultConfig(user2, partitionValue)
+        Realm.getInstance(config2).use { realm ->
+            assertEquals(0, realm.where<EmbeddedSimpleListParent>().count())
+            assertEquals(0, realm.where<EmbeddedSimpleChild>().count())
+
+            realm.syncSession.downloadAllServerChanges(5, TimeUnit.SECONDS).let {
+                if (!it) fail()
+            }
+            realm.refresh()
+
+            val childResults = realm.where<EmbeddedSimpleChild>()
+            assertEquals(2, childResults.count())
+            val parentResults = realm.where<EmbeddedSimpleListParent>()
+            assertEquals(1, parentResults.count())
+            val parentFromResults = parentResults.findFirst()!!
+            assertEquals(primaryKeyValue, parentFromResults._id)
+        }
+    }
+
+    @Test
+    fun embeddedObject_realmList_copyUnmanaged_roundTrip() {
+        val user1: User = createNewUser()
+        val config1: SyncConfiguration = createDefaultConfig(user1, partitionValue)
+        val primaryKeyValue = UUID.randomUUID().toString()
+        Realm.getInstance(config1).use { realm ->
+            realm.executeTransaction {
+                EmbeddedSimpleListParent(primaryKeyValue).apply {
+                    children = RealmList(EmbeddedSimpleChild("child1"), EmbeddedSimpleChild("child2"))
+                    it.insert(this)
+                }
+            }
+            realm.syncSession.uploadAllLocalChanges()
+
+            assertEquals(1, realm.where<EmbeddedSimpleListParent>().count())
+            assertEquals(2, realm.where<EmbeddedSimpleChild>().count())
+        }
+
+        val user2: User = createNewUser()
+        val config2: SyncConfiguration = createDefaultConfig(user2, partitionValue)
+        Realm.getInstance(config2).use { realm ->
+            assertEquals(0, realm.where<EmbeddedSimpleListParent>().count())
+            assertEquals(0, realm.where<EmbeddedSimpleChild>().count())
+
+            realm.syncSession.downloadAllServerChanges(1, TimeUnit.SECONDS).let {
+                if (!it) fail()
+            }
+            realm.refresh()
+
+            val childResults = realm.where<EmbeddedSimpleChild>()
+            assertEquals(2, childResults.count())
+            val parentResults = realm.where<EmbeddedSimpleListParent>()
+            assertEquals(1, parentResults.count())
+            val parentFromResults = parentResults.findFirst()!!
+            assertEquals(primaryKeyValue, parentFromResults._id)
+            assertEquals("child1", childResults.findAll()[0]!!.childId)
+            assertEquals("child2", childResults.findAll()[1]!!.childId)
+        }
+    }
+
+    @Test
+    @Ignore("Enable when a bug server-side is fixed")
```

Comment:
Do we have a reference to the issue?

Reply:
I've just got feedback from Nikola, I'll update when creating a ticket for them 👍